src/core/HistoryHandler.ts

```diff
@@ -48,8 +52,8 @@ export class HistoryHandler extends EventTarget {
 		naja.uiHandler.addEventListener('interaction', this.configureMode.bind(this));
 
 		this.historyAdapter = {
-			replaceState: (state, url) => window.history.replaceState(state, '', url),
-			pushState: (state, url) => window.history.pushState(state, '', url),
+			replaceState: (state, title, url) => window.history.replaceState(state, title, url),
+			pushState: (state, title, url) => window.history.pushState(state, title, url),
```

Comment:
We shouldn't do this. The "title" parameter is actually [`unused`](https://html.spec.whatwg.org/multipage/nav-history-apis.html#the-history-interface), Safari is the only browser that has ever supported this, and [that will hopefully end soon](https://github.com/WebKit/WebKit/commit/443a65a4c3a25485b3083c8a92f3be46e71102f2).

This works in Safari as well:

```javascript
history.pushState(state, '', url)
document.title = title
```

Reply:
I tested this quite thoroughly in the latest Safari 17.5 (19618.2.12.11.6). While you're right that an empty string works well in a newly pushed state, I've run into an edge case in replaceInitialState: passing an empty string leads to the first history entry having no title at all and displaying the page URL instead.

I'll be more than happy to remove this (again) once Safari 18 is out for some time, but for now, this works more reliably in Safari, without affecting other browsers.

Reply:
Oh. I just realized that I don't deal with this (the default/initial history "label" in Safari) anymore. You're right, sorry for the nitpicking 😄

Reply:
No problem, thanks for taking a look at the code :)